emop-maven-plugin/src/main/java/edu/cornell/VmsMojo.java

```diff
@@ -201,66 +385,57 @@ private boolean isRenamed(String oldClass, String newClass) {
     }
 
     /**
-     * Determines whether an old line in a file can be mapped to the new line. An offset is calculated which determines
-     * the maximum boundary of the distance a line has moved based off of differences between code versions. If two
-     * lines are within this offset of each other (in the correct direction), they are considered mappable.
-     * Take the following example:
+     * Determines whether an old line in a file can be mapped to the new line. For an old line to be mappable to a new
+     * line in a class, it must fulfill two conditions: that the old line has not been modified and that the old line
+     * in addition to the net offset of previous edits equals the new line.
+     * Take the following example where the second line of code has been modified with a new line inserted directly
+     * after it:
      * Old code          New code
-     * line 1            line 1
-     * line 2            line 2
-     * line 3            new line
-     * line 4            line 3
-     *                   line 4
-     * Lines 1 and 2 do not see any differences, so their offset is 0 and they will only map to lines 1 and 2 in the new
-     * code, respectively. Line 3 sees an addition of one line, so the offset here is 1 and both the new line and line 3
-     * in the new code will map to the previous line 3. Similarly, line 4 also takes the previous addition of a line
-     * into account and has an offset of 1, and both lines 3 and 4 in the new code will map to the previous line 4.
-     * Deletions work similarly, and are represented by a negative offset.
-     * TODO: The current implementation is generous with finding the "same" violation. The offset is very accurate in
-     *  finding the precise location unmodified code has been moved to. This information can be leveraged for more
-     *  accuracy when differentiating which violations are new or old when they occur close together and the line
-     *  itself where the violation occurred is not a part of any direct differences in code. There are also errors that
-     *  can arise because of differences between the last commit and the previous run of violation-counts (whose
-     *  violations we keep track of) this can be fixed by incorporating sha information.
+     * 1 line A          1 line A
+     * 2 line B          2 line B'
+     * 3 line C          3 new line
+     *                   4 line C
+     * The first line of code does not have any edits occurring earlier in the code, and will be matched directly to the
+     * first line of the new version of code. The second line has been modified and will not be mapped. The third line
+     * of code has an edit occurring earlier in the code with an offset of 1. Therefore, the third line of the old code
+     * will be mapped to the fourth line of the new code.
```

Comment:
what do you do if there is a violation on the new line 3?

Reply:
No line of the old code will ever be mapped to the newly-inserted line, so therefore no previous violation will ever map to a violation occurring on the new line 3 and the violation will be reported as "new".
I updated the description so it's a bit more clear.